### PR TITLE
add(main): new package alist

### LINKED ISSUE #9851: [Package]: alist
### Package description

A file list program that supports multiple storage, powered by Gin and React.

### Home page URL

https://alist-doc.nn.ci/

### Source code URL

https://github.com/Xhofe/alist

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by 2096779623**
Closes #9851

### CODE REVIEW

File: packages/alist/build.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
+TERMUX_PKG_HOMEPAGE=https://alist-doc.nn.ci
+TERMUX_PKG_DESCRIPTION="A file list program that supports multiple storage."
+TERMUX_PKG_LICENSE="AGPL-V3"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=2.5.2
+TERMUX_PKG_SRCURL=https://github.com/Xhofe/alist/archive/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=2f62ede85e9cabb3a338e2274301ab0389a2e9d986b32247ef9d8b48cec481b0
+TERMUX_PKG_BUILD_DEPENDS="yarn,git,golang (>= 1.17)"
```
Comment (MrAdityaAlok) [inline packages/alist/build.sh:8]: Why `git`?

File: packages/alist/build.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
+TERMUX_PKG_HOMEPAGE=https://alist-doc.nn.ci
+TERMUX_PKG_DESCRIPTION="A file list program that supports multiple storage."
+TERMUX_PKG_LICENSE="AGPL-V3"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=2.5.2
```
Comment (MrAdityaAlok) [inline packages/alist/build.sh:5]: Update to 2.6.0

**Review by AstralFetus84 — COMMENTED** [review]
Very well layed out

**Comment by 2096779623** [comment]
@MrAdityaAlok Please add this:
```bash
termux_step_pre_configure(){
# Switch to local to prevent some countries from not connecting to cdn.
        sed -i 's/https:\/\/npm.*\/dist/\//g' "$TERMUX_PKG_SRCDIR"/alist/conf/config.go
}
```

**Comment by MrAdityaAlok** [comment]
> @MrAdityaAlok Please add this:
> 
> ```shell
> termux_step_pre_configure(){
> # Switch to local to prevent some countries from not connecting to cdn.
>         sed -i 's/https:\/\/npm.*\/dist/\//g' "$TERMUX_PKG_SRCDIR"/conf/config.go
> }
> ```

Why? Isn't https://npm.elemecdn.com accessible from China (I guess, most probably that's your country)?

`alist` does not serve local files. Replacing that will simply break things.

**Comment by 2096779623** [comment]
> > @MrAdityaAlok Please add this:
> > ```shell
> > termux_step_pre_configure(){
> > # Switch to local to prevent some countries from not connecting to cdn.
> >         sed -i 's/https:\/\/npm.*\/dist/\//g' "$TERMUX_PKG_SRCDIR"/conf/config.go
> > }
> > ```
> 
> Why? Isn't https://npm.elemecdn.com accessible from China (I guess, most probably that's your country)?
> 
> `alist` does not serve local files. Replacing that will simply break things.

Using local files is faster than any cdn, in China some areas cannot access this cdn because they are blocked by GFW.

**Comment by MrAdityaAlok** [comment]
> 

Ok, then I'll try something.

**Comment by MrAdityaAlok** [comment]
> `alist` does not serve local files. Replacing that will simply break things.

Ahh..I was wrong here. It does server local files. It embeds all files in `public/` dir to binary.

**Pull request merged**